app/utils.js

```diff
@@ -129,6 +129,73 @@ define(['underscore'], function (_) {
     return value == null || value === '';
   };
 
+  /**
+   * Returns a function, that, as long as it continues to be invoked, will not be triggerd.
+   * The function will be called after it stops being called for N ms.
+   * Based on https://github.com/component/debounce/blob/master/index.js which in turn is based on the
+   * original implemenation in Underscore (both licensed MIT)
+   */
+  Utils.debounce = function(func, wait, immediate) {
```

Comment:
We use a lot of underscore utils over the place and it's already loaded so it won't make much pain.

We avoid to have to functions doing the same thing, and underscore will be proxy/wrapped by Utils some day.

You can add a function inside the interface so when it's time to move it, if, then add the code in there and should be good to go.

Or, use the underscore one inside the Utils, and that should be good to go too.